src/FormElement/ScheduleElement/MonthlyFields.php

```diff
@@ -86,6 +86,30 @@ public function getSelectedDays(): array
         return $selectedDays;
     }
 
+    public function validate()
+    {
+        parent::validate();
+        if (! $this->isValid || $this->getPopulatedValue('runsOn', static::RUNS_EACH) === static::RUNS_ONTHE) {
+            return $this;
+        }
+
+        $this->isValid = false;
+        foreach (range(1, $this->availableFields) as $day) {
```

Comment:
Can't this be done via `hasValue()`?

Reply:
No, otherwise it's not validated partially. (now)